doctests/dt_json.py

```diff
@@ -252,6 +264,19 @@
 assert res10 == ["Quaoar", "Weywot", "Salacia", "Mimas"]
 # REMOVE_END
 
+# STEP_START filter4
+res11 = r.json().set("bikes:inventory", "$.inventory.mountain_bikes[0].regex_pat", "(?i)al")
+res12 = r.json().set("bikes:inventory", "$.inventory.mountain_bikes[1].regex_pat", "(?i)al")
+res13 = r.json().set("bikes:inventory", "$.inventory.mountain_bikes[2].regex_pat", "(?i)al")
+
```

Comment:
`res11`, `res12` and `res13` are also used below in the original code. If possible it would be cleaner to not reuse names. Also add asserts for these, either here, or below (only `res12` is asserted there).